components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/PAPStatusDataHandlerTest.java

```diff
@@ -0,0 +1,330 @@
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.entitlement.EntitlementException;
+import org.wso2.carbon.identity.entitlement.PAPStatusDataHandler;
+import org.wso2.carbon.identity.entitlement.PDPConstants;
+import org.wso2.carbon.identity.entitlement.dto.PublisherDataHolder;
+import org.wso2.carbon.identity.entitlement.dto.PublisherPropertyDTO;
+import org.wso2.carbon.identity.entitlement.dto.StatusHolder;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementConfigHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import static org.mockito.Mockito.mockStatic;
+import static org.testng.Assert.assertEquals;
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.PROP_USE_LAST_STATUS_ONLY;
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.PolicyPublish.ACTION_CREATE;
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.PolicyPublish.ACTION_DELETE;
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.PolicyPublish.ACTION_UPDATE;
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.StatusTypes.ADD_POLICY;
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.StatusTypes.DELETE_POLICY;
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.StatusTypes.GET_POLICY;
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.StatusTypes.PUBLISH_POLICY;
+import static org.wso2.carbon.identity.entitlement.persistence.SubscriberPersistenceManagerTest.SUBSCRIBER_MODULE_NAME;
+
+/**
+ * This is the parent test class for the PAP Status Data Handler test classes.
+ */
+public abstract class PAPStatusDataHandlerTest {
+
+    static final String SUBSCRIBER_ID_KEY = "subscriberId";
+    static final String SUBSCRIBER_ID_DISPLAY_NAME = "Subscriber Id";
+
+    protected static final String ABOUT_POLICY = "POLICY";
+    protected static final String ABOUT_SUBSCRIBER = "SUBSCRIBER";
+    protected static final String POLICY_KEY = "simple_policy";
+    protected static final String SUBSCRIBER_KEY = "PDP_Subscriber_test";
+    protected static final String PAP_POLICY_STORE_TARGET = "PAP POLICY STORE";
+    protected static final String TARGET_ACTION_PERSIST = "PERSIST";
+    protected static final String TARGET_ACTION_REMOVE = "REMOVE";
+    protected static final String TARGET_ACTION_LOAD = "LOAD";
+    protected static final String POLICY_VERSION_1 = "1";
+    protected static final String SAMPLE_USER = "admin";
+
+    StatusHolder statusHolderForAddPolicy;
+    StatusHolder statusHolderForGetPolicy;
+    StatusHolder statusHolderForRemovePolicy;
+
+    StatusHolder statusHolderForAddPolicyInPDP;
+    StatusHolder statusHolderForUpdatePolicyInPDP;
+    StatusHolder statusHolderForRemovePolicyInPDP;
+
+    List<StatusHolder> policyStatusHolderList1;
```

Comment:
Shall we use more appropriate variable names that indicate the usage of it?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/compare/e393b967246d88953a960b77a0c4a7c355fd9dfd..f26be27fb80abcc3adb9aace8db0909c0eb3c17d